### PR TITLE
Support new show functions syntax to make user search function more conveniently

### PR DESCRIPTION
**Pull request by caiconghui**
Support new Show functions syntax to make functions more friendly for user to use

### CODE REVIEW

**Comment by caiconghui** [comment]
#2795

File: fe/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 1]
```diff
@@ -302,44 +304,77 @@ private void handleShowEngines() {
...
+            // like predicate
+            if (showStmt.getWild() == null || showStmt.like(function.functionName())) {
+                rowSet.add(row);
             }
-            // property
-            row.add(function.getProperties());
-            rowSet.add(row);
         }
+
+        // sort function rows by fisrt column asec
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/qe/ShowExecutor.java]:
```suggestion
        // sort function rows by first column asc
```

File: fe/src/main/java/org/apache/doris/qe/ShowExecutor.java [hunk 2]
```diff
@@ -302,44 +304,77 @@ private void handleShowEngines() {
...
-                row.add("Scalar");
-                row.add("NULL");
-            } else {
-                row.add("Aggregate");
-                AggregateFunction aggFunc = (AggregateFunction) function;
-                Type intermediateType = aggFunc.getIntermediateType();
-                if (intermediateType != null) {
-                    row.add(intermediateType.getPrimitiveType().toString());
-                } else {
+            List<Comparable> row = Lists.newArrayList();
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/qe/ShowExecutor.java]: I think the `row` built here can be implemented in `Function`. eg:

function.getInfo(row, isVerbose);

File: fe/src/main/cup/sql_parser.cup [hunk 3]
```diff
@@ -194,14 +194,14 @@ parser code {:
 
 // Total keywords of doris
 terminal String KW_ADD, KW_ADMIN, KW_AFTER, KW_AGGREGATE, KW_ALL, KW_ALTER, KW_AND, KW_ANTI, KW_AS, KW_ASC, KW_AUTHORS, 
-    KW_BACKEND, KW_BACKUP, KW_BETWEEN, KW_BEGIN, KW_BIGINT, KW_BITMAP, KW_BITMAP_UNION, KW_BOOLEAN, KW_BOTH, KW_BROKER, KW_BACKENDS, KW_BY,
+    KW_BACKEND, KW_BACKUP, KW_BETWEEN, KW_BEGIN, KW_BIGINT, KW_BITMAP, KW_BITMAP_UNION, KW_BOOLEAN, KW_BOTH, KW_BROKER, KW_BACKENDS, KW_BY, KW_BUILTIN,
```
Comment (morningman) [inline fe/src/main/cup/sql_parser.cup]: Add newly added keyword to `keyword ::=`

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**